keyboards/ut472/keymaps/praxling/keymap.c

```diff
@@ -0,0 +1,283 @@
+	* `-------------------------------------------------------------------------'
+	*/
+	[3] = LAYOUT(
+		KC_ESC, KC_COLN, KC_COMM, KC_EQL, KC_NO, KC_NO, KC_NO, KC_P7, KC_P8, KC_P9, KC_NO, KC_BSPC, 
+		KC_TRNS, KC_NO, KC_PMNS, KC_PPLS, KC_NO, KC_NO, KC_NO, KC_P4, KC_P5, KC_P6, KC_PAST, KC_PSLS, 
+		KC_TRNS, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_PDOT, KC_P1, KC_P2, KC_P3, KC_NO, KC_SFTENT, 
+		KC_TRNS, RGB_TOG, RGB_RMOD, RGB_MOD, KC_TRNS, KC_TRNS, KC_P0, KC_LEFT, KC_DOWN, KC_UP, KC_RGHT),
+
+
+    /* FN Layer 4
+  * ,-------------------------------------------------------------------------.
+  * | Esc |ScrlDn|ScrlUp|Mous1|Mous2|    |   |MousL|MousD|MousU|MousR| Bspace |
+  * |-------------------------------------------------------------------------+
+  * | Hyp | Shift | Ctrl | Alt | Gui |   |   | Left | Down | Up | Right | Del |
+  * |-------------------------------------------------------------------------+
+  * |        | Undo | Cut | Copy | Paste |     |     |Home|PgDn|PgUp|End| Ent |
+  * |-------------------------------------------------------------------------+
+  * |     |    |    |    |       |            |       |      |     |    |     |
+  * `-------------------------------------------------------------------------'
+  */
+  [4] = LAYOUT(
+    KC_ESC, KC_NO, KC_SCRL_DWN, KC_SCRL_UP, KC_BTN1, KC_BTN2, KC_NO, KC_MS_L, KC_MS_D, KC_MS_U, KC_MS_R, KC_BSPC, 
+    KC_HYP, KC_LSFT, KC_LCTL, KC_LALT, KC_LGUI, KC_NO, KC_NO, KC_LEFT, KC_DOWN, KC_UP, KC_RIGHT, KC_DEL, 
+    KC_TRNS, UC_UNDO, UC_CUT, UC_COPY, UC_PASTE, KC_NO, KC_NO, KC_HOME, KC_PGDN, KC_PGUP, KC_END, KC_ENT, 
+    KC_TRNS, KC_NO, KC_NO, KC_NO, KC_TRNS, KC_TRNS, KC_TRNS, KC_NO, KC_NO, KC_NO, KC_NO),
+
+
+  /* FN Layer 5 (for switching virtual desktops)
+  * ,-------------------------------------------------------------------------.
+  * |     |     |     |     |     |     |     |     | Left |Right|     |      |
+  * |-------------------------------------------------------------------------+
+  * |      |  1  |  2  |  3  |  4  |  5  |  6  |  7  |  8  |  9  |  0  |      |
+  * |-------------------------------------------------------------------------+
+  * |       |     |     |     |     |     |     |     |     |     |     |     |
+  * |-------------------------------------------------------------------------+
+  * |     |    |    |    |        |              |        |    |    |    |    |
+  * `-------------------------------------------------------------------------'
+  */
+  [5] = LAYOUT(
+    KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_LEFT, KC_RGHT, KC_NO, KC_NO, 
+    KC_NO, KC_P1, KC_P2, KC_P3, KC_P4, KC_P5, KC_P6, KC_P7, KC_P8, KC_P9, KC_P0, KC_NO, 
+    KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, 
+    KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO),  
+};
+
+
+
+//Custom functions
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```

Comment:
To be honest, you can probably remove this ENTIRE function.

It looks like mostly what it's doing is swapping CTRL and GUI/CMD keys.   Since that's the case .... you could use `C(kc)` (aka `LCTL(kc)`), and use the Control-GUI swap "magic" feature:
https://docs.qmk.fm/#/feature_bootmagic?id=keycodes

Specifically, the `CG_SWAP` and `CG_NORM`, or `CG_TOGG` keycodes.

Reply:
I think you might be right.
I originally opted to write my own function so that the locations of the modifier keys would be consistent across OS, since the swap keycodes globally swaps the locations where they're mapped. Now that I'm more familiar with QMK and custom layouts in general, I realize that the latter is what I want so that I'm not fighting with muscle memory.

The only incompatibility I see right off the bat is my custom keycode for locking the screen, but I think I have a workaround for that.

Reply:
Well, for a lot of these, you could use:
```c
register_mod(mod_config(MOD_BIT(KC_LCTRL)));
tap_code(KC_x);
unregister_mod(mod_config(MOD_BIT(KC_LCTRL)));
```

This would look at the Control-GUI swap settings, and other settings and change the control to gui, if needed.  

However, really, you could just use `C(KC_x)` in the keymap and it will basically do the same thing.